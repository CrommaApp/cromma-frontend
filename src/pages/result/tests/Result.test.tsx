import React from 'react';
import { screen, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Result from '@pages/result';
import SearchNewsService from '@services/search/search-news';
import { MemoryRouter, Route } from 'react-router';
import { BrowserRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

describe('Result', () => {
	let searchNewsService: SearchNewsService;

	const clickNextButton = () => {
		const nextButton = screen.getByRole('button', {
			name: 'next',
		});

		userEvent.click(nextButton);
	};

	const clickPrevButton = () => {
		const prevButton = screen.getByRole('button', {
			name: 'prev',
		});

		userEvent.click(prevButton);
	};

	beforeEach(() => {
		searchNewsService = new SearchNewsService();

		render(
			<MemoryRouter initialEntries={['/result/korea']}>
				<Route exact path="/result/:keyword">
					<Result searchNewsService={searchNewsService} />
				</Route>
			</MemoryRouter>,
		);
	});
	it('Snapshot', () => {
		const component = renderer.create(
			<BrowserRouter>
				<Result searchNewsService={searchNewsService} />
			</BrowserRouter>,
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Show keyword in the parameter', () => {
		expect(
			screen.getByRole('heading', {
				name: 'Search results for korea.',
			}),
		).toBeInTheDocument();
	});
});
