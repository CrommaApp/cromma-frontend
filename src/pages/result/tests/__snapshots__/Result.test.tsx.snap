// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Snapshot 1`] = `
<section
  className="sc-hKwDye hOYbwV"
>
  <h1
    id="search_result"
  >
    Search results for 
    korea
    .
  </h1>
  <ul
    aria-labelledby="search_result"
    className="sc-gsDKAQ zvsYr"
  />
  <div
    className="sc-eCImPb fgkKjW"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      prev
    </button>
    <p
      data-testid="current_page"
    >
      1
    </p>
    <button
      onClick={[Function]}
      type="button"
    >
      next
    </button>
  </div>
</section>
`;
